mass_mailing_custom_unsubscribe/controllers/main.py

```diff
@@ -2,13 +2,32 @@
 # Copyright 2016 Tecnativa - Jairo Llopis
 # Copyright 2020 Tecnativa - Pedro M. Baeza
 # Copyright 2024 Tecnativa - David Vidal
+# Copyright 2025 Moduon Team
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 from odoo.http import request, route
 
 from odoo.addons.mass_mailing.controllers.main import MassMailController
 
 
 class CustomUnsubscribe(MassMailController):
+    @route()
+    def mailing_confirm_unsubscribe(
+        self, mailing_id, document_id=None, email=None, hash_token=None
+    ):
+        """Redirect to the unsubscribe page if the redirect option is enabled."""
+        if (
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("mass_mailing.mailing_lists_unsubscribe_redirect", "0")
+            == "1"
+        ):
```

Comment:
For booleans, it's simpler to just check if the param is there or not (and that's how it will work if you use it as a `res.config.settings` boolean `config_parameter`
```suggestion
        if (
            request.env["ir.config_parameter"]
            .sudo()
            .get_param("mass_mailing.mailing_lists_unsubscribe_redirect")
        ):
```

Reply:
Actually you have `odoo.tools.str2bool` for this specific purpose.


```suggestion
        if str2bool(
            request.env["ir.config_parameter"]
            .sudo()
            .get_param("mass_mailing.mailing_lists_unsubscribe_redirect", "0")
        ):
```